peer/common/deliverclient.go

```diff
@@ -71,11 +71,16 @@ func (d *DeliverClient) readBlock() (*cb.Block, error) {
 	}
 	switch t := msg.Type.(type) {
 	case *ab.DeliverResponse_Status:
-		logger.Infof("Got status: %v", t)
+		logger.Infof("Expect block, but got status: %v", t)
 		return nil, errors.Errorf("can't read the block: %v", t)
 	case *ab.DeliverResponse_Block:
 		logger.Infof("Received block: %v", t.Block.Header.Number)
-		d.Service.Recv() // Flush the success message
+		if resp, err := d.Service.Recv(); err != nil { // Flush the success message
+			logger.Errorf("Failed to flush success message: %s", err)
```

Comment:
This is not the log message I was expecting to see here, but it's a nit admittedly, and I see your point about this being a means to an end.

Reply:
updated to be more precise, hopefully we could spot the ghost and get rid of these "debugging" code